doc/how_to/custom_components/esm/panes.md

```diff
@@ -0,0 +1,178 @@
+            "datasets": [
+                {
+                    "label": "Data",
+                    "backgroundColor": "rgb(255, 99, 132)",
+                    "borderColor": "rgb(255, 99, 132)",
+                    "data": [0, 10, 5, 2, 20, 30, 45],
+                }
+            ],
+        },
+        "options": {
+            "responsive": True,
+            "maintainAspectRatio": False,
+        },
+    }
+
+
+chart_type = pn.widgets.RadioBoxGroup(
+    name="Chart Type", options=["bar", "line"], inline=True
+)
+chart = ChartJSComponent(
+    object=pn.bind(data, chart_type), height=400, sizing_mode="stretch_width"
+)
+pn.Column(chart_type, chart).servable()
+```
+
+Note the use of the `requestAnimationFrame` to postpone the rendering of the chart. If not used then Chart.js will not render. Dealing with layout issues like this sometimes requires a bit of iteration. If you get stuck, share your question and minimum, reproducible code example on [Discourse](https://discourse.holoviz.org/).
+
+## Creating a Cytoscape Pane
+
+This example will show you how to build a more advanced [CytoscapeJS](https://js.cytoscape.org/) pane.
+
+```{pyodide}
+import param
+import panel as pn
+
+from panel.custom import JSComponent
+
+
+class Cytoscape(JSComponent):
+
+    object = param.List()
+
+    layout = param.Selector(
+        default="cose",
+        objects=[
+            "breadthfirst",
+            "circle",
+            "concentric",
+            "cose",
+            "grid",
+            "preset",
+            "random",
+        ],
+    )
+    style = param.String("", doc="Use to set the styles of the nodes/edges")
+
+    zoom = param.Number(1, bounds=(1, 100))
+    pan = param.Dict({"x": 0, "y": 0})
+
+    data = param.List(doc="Use to send node's data/attributes to Cytoscape")
+
+    selected_nodes = param.List()
+    selected_edges = param.List()
+
+    _esm = """
+import { default as cytoscape} from "https://esm.sh/cytoscape"
+let cy = null;
+
+export function render({ model, el }) {
+    if (cy) { cy.destroy() }
+
+    cy = cytoscape({
+        container: el,
+        layout: {name: model.layout},
+        elements: model.object,
+        zoom: model.zoom,
+        pan: model.pan
+    })
+    cy.on('select unselect', function (evt) {
+        model.selected_nodes = cy.elements('node:selected').map(el => el.id())
+        model.selected_edges = cy.elements('edge:selected').map(el => el.id())
+    });
+    cy.style().resetToDefault().append(model.style).update()
+
+    model.on('object', () => {cy.json({elements: model.object});cy.resize().fit()})
+    model.on('layout', () => {cy.layout({name: model.layout}).run()})
+    model.on('zoom', () => {cy.zoom(model.zoom)})
+    model.on('pan', () => {cy.pan(model.pan)})
+    model.on('style', () => {cy.style().resetToDefault().append(model.style).update()})
+
+    window.addEventListener('resize', function(event){
```

Comment:
What is the correct and recommended way to trigger responsive resizing?

Reply:
`model.on('resize', ...)` or more explicitly `model.on('lifecycle:resize', ...)`

Reply:
technically `model.on('after_resize', function(event){...` works. But the redraw is not very smooth

https://github.com/user-attachments/assets/ab286890-3613-4dd4-aae4-6837c27257b7

Instead `window.addEventListener('resize', function(event){...` works much smoother

https://github.com/user-attachments/assets/d90daa44-1896-4143-8a13-f4fa8a8fc1d0

I would recommend using the last option. Is there any way to get the first option working properly @philippjfr ?

Reply:
In my current testing `after_resize` works just the same. I think we should rename it to `model.on('resize', ...)` though.